### PR TITLE
increase docupdater logging

### PR DESCRIPTION
**Pull request by briangough**
log all lock operations and also log slow update operations

### CODE REVIEW

File: app/coffee/LockManager.coffee [hunk 1]
```diff
@@ -30,14 +30,16 @@ module.exports = LockManager =
 	tryLock : (doc_id, callback = (err, isFree)->)->
 		lockValue = LockManager.randomLock()
 		key = keys.blockingKey(doc_id:doc_id)
+		logger.log {doc_id: doc_id, key: key}, "tryLock"
```
Comment (jpallen) [inline app/coffee/LockManager.coffee]: These are going to be VERY noisy, probably too much so.

File: app/coffee/LockManager.coffee [hunk 2]
```diff
@@ -30,14 +30,16 @@ module.exports = LockManager =
...
 		rclient.set key, lockValue, "EX", @LOCK_TTL, "NX", (err, gotLock)->
 			return callback(err) if err?
 			if gotLock == "OK"
 				metrics.inc "doc-not-blocking"
+				logger.log {doc_id: doc_id, key: key, lockValue: lockValue}, "got lock"
```
Comment (jpallen) [inline app/coffee/LockManager.coffee]: Ditto

File: app/coffee/LockManager.coffee [hunk 3]
```diff
@@ -75,7 +78,9 @@ module.exports = LockManager =
...
 			if err?
 				return callback(err)
-			if result? and result isnt 1 # successful unlock should release exactly one key
-				logger.error {doc_id:doc_id, lockValue:lockValue, redis_err:err, redis_result:result}, "unlocking error"
+			else if result? and result isnt 1 # successful unlock should release exactly one key
+				logger.error {doc_id:doc_id, key:key, lockValue:lockValue, redis_err:err, redis_result:result}, "unlocking error"
 				return callback(new Error("tried to release timed out lock"))
-			callback(err,result)
+			else
+				logger.log {doc_id:doc_id, key:key, lockValue:lockValue}, "released lock"
```
Comment (jpallen) [inline app/coffee/LockManager.coffee]: Ditto

File: app/coffee/Profiler.coffee [hunk 4]
```diff
@@ -0,0 +1,28 @@
+Settings = require('settings-sharelatex')
+logger = require('logger-sharelatex')
+
+deltaMs = (ta, tb) ->
+	return Math.floor(((ta[0]-tb[0])*1e9 + (ta[1]-tb[1]))*1e-6)
+
+module.exports = class Profiler
+	LOG_CUTOFF_TIME: 100
```
Comment (jpallen) [inline app/coffee/Profiler.coffee]: Could we increase this to bring down the noise?

File: app/coffee/Profiler.coffee [hunk 5]
```diff
@@ -0,0 +1,28 @@
+Settings = require('settings-sharelatex')
+logger = require('logger-sharelatex')
+
+deltaMs = (ta, tb) ->
+	return Math.floor(((ta[0]-tb[0])*1e9 + (ta[1]-tb[1]))*1e-6)
```
Comment (jpallen) [inline app/coffee/Profiler.coffee]: Can we make this a little more readable, perhaps:

```
nanoseconds = (ta[0]-tb[0])*1e9 + (ta[1]-tb[1])
milliseconds = Math.floor(nanoseconds *1e-6)
```

It took me a while to understand as is

**Comment by briangough** [comment]
changed it all to use the profiler and only report operations taking more than 1s - hopefully negligible impact on the logs

**Pull request merged**